src/pagination/pagination.ts

```diff
@@ -48,46 +48,31 @@ import {getValueInRange, toInteger, toBoolean} from '../util/util';
   `
 })
 export class NgbPagination implements OnChanges {
-  private _boundaryLinks = false;
-  private _collectionSize;
-  private _directionLinks = true;
-  private _ellipses = true;
-  private _maxSize = 0;
-  private _page = 0;
-  private _pageCount = 0;
-  private _pageSize = 10;
-  private _rotate = false;
-  pages: number[] = [];
-
   /**
    *  Whether to show the "First" and "Last" page links
    */
-  @Input()
-  set boundaryLinks(value: boolean) {
-    this._boundaryLinks = toBoolean(value);
-  }
-
-  get boundaryLinks(): boolean { return this._boundaryLinks; }
+  @Input() boundaryLinks = false;
 
+  private _collectionSize;
   /**
    *  Whether to show the "Next" and "Previous" page links
    */
-  @Input()
-  set directionLinks(value: boolean) {
-    this._directionLinks = toBoolean(value);
-  }
-
-  get directionLinks(): boolean { return this._directionLinks; }
+  @Input() directionLinks = true;
 
   /**
    *  Whether to show ellipsis symbols and first/last page numbers when maxSize > number of pages
    */
-  @Input()
-  set ellipses(value: boolean) {
-    this._ellipses = toBoolean(value);
-  }
-
-  get ellipses(): boolean { return this._ellipses; }
+  @Input() ellipses = true;
+  private _maxSize = 0;
+  private _page = 0;
+  private _pageCount = 0;
+  private _pageSize = 10;
```

Comment:
I agree, but the idea of this PR was to only focus on boolean inputs, and to keep numbers as is for the moment, mainly because I wasn't sure what to do with all these number inputs.

I will definitely do another PR to deal with numbers, but I wasn't sure about what to do with all the number inputs here. 

We agreed that `number | string` should become `number`, but the usage of `toInteger()` here has another effect: it transforms decimal numbers to integers. And I am not sure that this is desired or not, or if we should just accept the user inputs as is.

Once I have an answer for that question (sorry if that wasn't clear in the issue), I will make another PR for numbers, if you don't mind.